fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/model/IDatabaseObject.java

```diff
@@ -25,21 +25,29 @@
      * @return
      */
     public int getVersion();
-    
+
     /**
      * Apply the DDL for this object to the target database
+     * @param priorVersion
      * @param target the database target
      */
     public void apply(IDatabaseAdapter target);
-    
+
+    /**
+     * Apply migration logic to bring the target database to the current level of this object
+     * @param priorVersion
+     * @param target the database target
+     */
+    public void apply(Integer priorVersion, IDatabaseAdapter target);
```

Comment:
I'd say add a NO OP default here.  It'll be fairly repeatable for StoredProcedures or other NO OP content such as Tablespaces.

Reply:
didn't end up doing this one but can if you think its important.
instead, i opted to add "ThrowUnsupportedOperationException" in cases where we're trying to update objects that we shouldn't (i.e. tablespace and sequence).